ffi struct produces warnings on Rust 1.79.0 or +nightly

I have a struct that I use to convert wrapped errors to strings that can be sent to Swift:

``` rust
#[swift_bridge::bridge]
mod ffi {
  #[swift_bridge(swift_repr = "struct")]
  pub struct BridgedError {
    pub inner: String
  }
}
```

This compiles without any errors in Rust 1.77.2, but when I compile it with nightly it produces the following warning:

```
> cargo +nightly build -Zbuild-std --package my-bridge --target aarch64-apple-darwin

error: field `0` is never read
  --> rust/my-bridge/src/lib.rs:17:14
   |
14 | #[swift_bridge::bridge]
   | ----------------------- field in this variant
...
17 |   pub struct BridgedError {
   |              ^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> rust/my-bridge/src/lib.rs:2:38
   |
2  | #![cfg_attr(feature = "strict", deny(warnings))]
   |                                      ^^^^^^^^
   = note: `#[deny(dead_code)]` implied by `#[deny(warnings)]`
help: consider changing the field to be of unit type to suppress this warning while preserving the field numbering, or remove the field
   |
17 |   pub struct () {
   |              ~~
```

I'm starting to play around with cross-compiling to watchOS, tvOS, and eventually visionOS, but am finding these targets only seem to build on nightly.

Also on nightly some of the `ui` tests fail to run.

<details>
<summary>Output</summary>

```
test tests/ui/args-into-argument-not-found.rs ... mismatch

EXPECTED:
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error: Argument "arg_typo" was not found in "fn some_function(..)"
 --> tests/ui/args-into-argument-not-found.rs:7:42
  |
7 |         #[swift_bridge(args_into = (arg, arg_typo))]
  |                                          ^^^^^^^^

error: Argument "bar" was not found in "fn some_method(..)"
  --> tests/ui/args-into-argument-not-found.rs:13:42
   |
13 |         #[swift_bridge(args_into = (foo, bar))]
   |                                          ^^^
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈

ACTUAL OUTPUT:
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error: Argument "arg_typo" was not found in "fn some_function(..)"
 --> tests/ui/args-into-argument-not-found.rs:7:42
  |
7 |         #[swift_bridge(args_into = (arg, arg_typo))]
  |                                          ^^^^^^^^

error: Argument "bar" was not found in "fn some_method(..)"
  --> tests/ui/args-into-argument-not-found.rs:13:42
   |
13 |         #[swift_bridge(args_into = (foo, bar))]
   |                                          ^^^

warning: unused variable: `arg`
  --> tests/ui/args-into-argument-not-found.rs:18:18
   |
18 | fn some_function(arg: u8) {}
   |                  ^^^ help: if this is intentional, prefix it with an underscore: `_arg`
   |
   = note: `#[warn(unused_variables)]` on by default

warning: unused variable: `foo`
  --> tests/ui/args-into-argument-not-found.rs:23:27
   |
23 |     fn some_method(&self, foo: u8) {}
   |                           ^^^ help: if this is intentional, prefix it with an underscore: `_foo`
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
note: If the actual output is the correct output you can bless it by rerunning
      your test with the environment variable TRYBUILD=overwrite

test tests/ui/incorrect-argument-type.rs ... mismatch

EXPECTED:
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error[E0308]: mismatched types
  --> tests/ui/incorrect-argument-type.rs:15:16
   |
10 | #[swift_bridge::bridge]
   | ----------------------- arguments to this function are incorrect
...
15 |         fn fn1(arg: &str);
   |                ^^^^^^ expected `u16`, found `&str`
   |
note: function defined here
  --> tests/ui/incorrect-argument-type.rs:24:4
   |
24 | fn some_function(_arg: u16) {}
   |    ^^^^^^^^^^^^^ ---------
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈

ACTUAL OUTPUT:
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error[E0308]: mismatched types
  --> tests/ui/incorrect-argument-type.rs:15:16
   |
10 | #[swift_bridge::bridge]
   | ----------------------- arguments to this function are incorrect
...
15 |         fn fn1(arg: &str);
   |                ^^^^^^^^^ expected `u16`, found `&str`
   |
note: function defined here
  --> tests/ui/incorrect-argument-type.rs:24:4
   |
24 | fn some_function(_arg: u16) {}
   |    ^^^^^^^^^^^^^ ---------
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
note: If the actual output is the correct output you can bless it by rerunning
      your test with the environment variable TRYBUILD=overwrite

test tests/ui/incorrect-return-type.rs ... mismatch

EXPECTED:
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error[E0308]: mismatched types
  --> tests/ui/incorrect-return-type.rs:6:1
   |
6  |   #[swift_bridge::bridge]
   |   ^^^^^^^^^^^^^^^^^^^^^^^ expected `SomeType`, found `&SomeType`
...
9  |           type SomeType;
   |  ______________-
10 | |
11 | |         #[swift_bridge(rust_name = "some_function")]
   | |_________- expected due to this
   |
   = note: this error originates in the attribute macro `swift_bridge::bridge` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/ui/incorrect-return-type.rs:6:1
   |
6  |   #[swift_bridge::bridge]
   |   ^^^^^^^^^^^^^^^^^^^^^^^ expected `SomeType`, found `Option<SomeType>`
...
9  |           type SomeType;
   |  ______________-
10 | |
11 | |         #[swift_bridge(rust_name = "some_function")]
12 | |         fn fn1() -> SomeType;
13 | |         #[swift_bridge(rust_name = "another_function")]
   | |_________- expected due to this
   |
   = note: expected struct `SomeType`
                found enum `Option<SomeType>`
   = note: this error originates in the attribute macro `swift_bridge::bridge` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider using `Option::expect` to unwrap the `Option<SomeType>` value, panicking if the value is an `Option::None`
   |
6  | #[swift_bridge::bridge].expect("REASON")
   |                        +++++++++++++++++
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈

ACTUAL OUTPUT:
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error[E0308]: mismatched types
  --> tests/ui/incorrect-return-type.rs:6:1
   |
6  |   #[swift_bridge::bridge]
   |   ^^^^^^^^^^^^^^^^^^^^^^^ expected `SomeType`, found `&SomeType`
...
9  |           type SomeType;
   |  ______________-
10 | |
11 | |         #[swift_bridge(rust_name = "some_function")]
12 | |         fn fn1() -> SomeType;
   | |_____________________________- expected due to this
   |
   = note: this error originates in the attribute macro `swift_bridge::bridge` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/ui/incorrect-return-type.rs:6:1
   |
6  |   #[swift_bridge::bridge]
   |   ^^^^^^^^^^^^^^^^^^^^^^^ expected `SomeType`, found `Option<SomeType>`
...
9  |           type SomeType;
   |  ______________-
10 | |
11 | |         #[swift_bridge(rust_name = "some_function")]
12 | |         fn fn1() -> SomeType;
13 | |         #[swift_bridge(rust_name = "another_function")]
14 | |         fn fn2() -> SomeType;
   | |_____________________________- expected due to this
   |
   = note: expected struct `SomeType`
                found enum `Option<SomeType>`
   = note: this error originates in the attribute macro `swift_bridge::bridge` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider using `Option::expect` to unwrap the `Option<SomeType>` value, panicking if the value is an `Option::None`
   |
6  | #[swift_bridge::bridge].expect("REASON")
   |                        +++++++++++++++++
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
note: If the actual output is the correct output you can bless it by rerunning
      your test with the environment variable TRYBUILD=overwrite

test tests/ui/invalid-associated-to-attribute.rs ... ok
test tests/ui/invalid-copy-attribute.rs ... ok
test tests/ui/invalid-module-item.rs ... ok
test tests/ui/opaque-copy-type-mut-ref.rs ... ok
test tests/ui/unrecognized-argument-attribute.rs ... ok
test tests/ui/unrecognized-enum-attribute.rs ... ok
test tests/ui/unrecognized-function-attribute.rs ... ok
test tests/ui/unrecognized-opaque-type-attribute.rs ... ok


test ui_tests::ui ... FAILED
```
</details>

--- a/SwiftRustIntegrationTestRunner/SwiftRustIntegrationTestRunner/Result.swift
+++ b/SwiftRustIntegrationTestRunner/SwiftRustIntegrationTestRunner/Result.swift
@@ -35,6 +35,9 @@ extension AsyncResultOpaqueRustType2: Error {}
 extension ResultTransparentEnum: @unchecked Sendable {}
 extension ResultTransparentEnum: Error {}
 
+extension ResultTransparentStruct: @unchecked Sendable {}
+extension ResultTransparentStruct: Error {}
+
 extension SameEnum: @unchecked Sendable {}
 extension SameEnum: Error {}
 

--- a/SwiftRustIntegrationTestRunner/SwiftRustIntegrationTestRunnerTests/ResultTests.swift
+++ b/SwiftRustIntegrationTestRunner/SwiftRustIntegrationTestRunnerTests/ResultTests.swift
@@ -193,6 +193,46 @@ class ResultTests: XCTestCase {
         }
     }
 
+    /// Verify that we can receive a Result<(), TransparentStruct> from Rust
+    func testResultNullTransparentStruct() throws {
+        try! rust_func_return_result_null_transparent_struct(true)
+
+        do {
+            try rust_func_return_result_null_transparent_struct(false)
+            XCTFail("The function should have returned an error.")
+        } catch let error as ResultTransparentStruct {
+            XCTAssertEqual(error.inner.toString(), "failed")
+        }
+
+        XCTContext.runActivity(named: "Should return a Unit type") {
+            _ in
+            do {
+                let _ :() = try rust_func_return_result_unit_type_enum_opaque_rust(true)
+            } catch {
+                XCTFail()
+            }
+        }
+
+        XCTContext.runActivity(named: "Should throw an error") {
+            _ in
+            do {
+                let _ :() = try rust_func_return_result_unit_type_enum_opaque_rust(false)
+                XCTFail("The function should have returned an error.")
+            } catch let error as ResultTransparentEnum {
+                switch error {
+                case .NamedField(let data):
+                    XCTAssertEqual(data, 123)
+                case .UnnamedFields(_, _):
+                    XCTFail()
+                case .NoFields:
+                    XCTFail()
+                }
+            } catch {
+                XCTFail()
+            }
+        }
+    }
+
     /// Verify that we can receive a Result<Vec<>, OpaqueRust> from Rust
     func testSwiftCallRustResultVecUInt32Rust() throws {
         let vec = try! rust_func_return_result_of_vec_u32()

--- a/crates/swift-bridge-ir/src/bridged_type/bridgeable_result.rs
+++ b/crates/swift-bridge-ir/src/bridged_type/bridgeable_result.rs
@@ -375,10 +375,14 @@ impl BuiltInResult {
             .err_ty
             .to_ffi_compatible_rust_type(swift_bridge_path, types);
         let mut custom_rust_ffi_types = vec![];
+        // TODO: remove allowances when rustc no longer issues dead code warnings for `#[repr(C)]`
+        //  structs or enums: https://github.com/rust-lang/rust/issues/126706
         custom_rust_ffi_types.push(quote! {
             #[repr(C)]
             pub enum #ty {
+                #[allow(unused)]
                 Ok #ok,
+                #[allow(unused)]
                 Err(#err),
             }
         });

--- a/crates/swift-bridge-ir/src/bridged_type/bridgeable_result.rs
+++ b/crates/swift-bridge-ir/src/bridged_type/bridgeable_result.rs
@@ -375,7 +375,7 @@ impl BuiltInResult {
             .err_ty
             .to_ffi_compatible_rust_type(swift_bridge_path, types);
         let mut custom_rust_ffi_types = vec![];
-        // TODO: remove allowances when rustc no longer issues dead code warnings for `#[repr(C)]`
+        // TODO: remove `#[allow(unused)]` when rustc no longer issues dead code warnings for `#[repr(C)]`
         //  structs or enums: https://github.com/rust-lang/rust/issues/126706
         custom_rust_ffi_types.push(quote! {
             #[repr(C)]

--- a/crates/swift-bridge-ir/src/codegen/codegen_tests/result_codegen_tests.rs
+++ b/crates/swift-bridge-ir/src/codegen/codegen_tests/result_codegen_tests.rs
@@ -408,7 +408,8 @@ mod extern_rust_fn_return_result_opaque_rust_type_and_transparent_enum_type {
     }
 
     // In Rust 1.79.0 dead_code warnings are issued for wrapped data in enums in spite of the enum
-    // having `#[repr(C)]`.
+    // having `#[repr(C)]`. `#[allow(unused)]` can be removed following resolution and release of this
+    // issue: https://github.com/rust-lang/rust/issues/126706
     fn expected_rust_tokens() -> ExpectedRustTokens {
         ExpectedRustTokens::Contains(quote! {
             #[repr(C)]
@@ -488,7 +489,9 @@ mod extern_rust_fn_return_result_transparent_enum_type_and_opaque_rust_type {
         }
     }
 
-    // Allows unused to avoid dead_code warnings in Rust 1.79.0 or later.
+    // In Rust 1.79.0 dead_code warnings are issued for wrapped data in enums in spite of the enum
+    // having `#[repr(C)]`. `#[allow(unused)]` can be removed following resolution and release of this
+    // issue: https://github.com/rust-lang/rust/issues/126706
     fn expected_rust_tokens() -> ExpectedRustTokens {
         ExpectedRustTokens::Contains(quote! {
             #[repr(C)]
@@ -565,7 +568,9 @@ mod extern_rust_fn_return_result_unit_type_and_transparent_enum_type {
         }
     }
 
-    // Allows unused to avoid dead_code warnings in Rust 1.79.0 or later.
+    // In Rust 1.79.0 dead_code warnings are issued for wrapped data in enums in spite of the enum
+    // having `#[repr(C)]`. `#[allow(unused)]` can be removed following resolution and release of this
+    // issue: https://github.com/rust-lang/rust/issues/126706
     fn expected_rust_tokens() -> ExpectedRustTokens {
         ExpectedRustTokens::Contains(quote! {
             #[repr(C)]
@@ -638,7 +643,9 @@ mod extern_rust_fn_return_result_tuple_type_and_transparent_enum_type {
         }
     }
 
-    // Allows unused to avoid dead_code warnings in Rust 1.79.0 or later.
+    // In Rust 1.79.0 dead_code warnings are issued for wrapped data in enums in spite of the enum
+    // having `#[repr(C)]`. `#[allow(unused)]` can be removed following resolution and release of this
+    // issue: https://github.com/rust-lang/rust/issues/126706
     fn expected_rust_tokens() -> ExpectedRustTokens {
         ExpectedRustTokens::ContainsMany(vec![
             quote! {

--- a/crates/swift-bridge-ir/src/codegen/codegen_tests/result_codegen_tests.rs
+++ b/crates/swift-bridge-ir/src/codegen/codegen_tests/result_codegen_tests.rs
@@ -707,3 +707,77 @@ typedef struct __swift_bridge__$ResultTupleI32U32AndSomeErrEnum{__swift_bridge__
         .test();
     }
 }
+
+/// Test code generation for Rust function that returns a Result<T, E> where T is () and
+/// E is a transparent struct type.
+mod extern_rust_fn_return_result_unit_type_and_transparent_struct_type {
+    use super::*;
+
+    fn bridge_module_tokens() -> TokenStream {
+        quote! {
+            mod ffi {
+                struct SomeErrStruct {
+                    inner: String,
+                }
+                extern "Rust" {
+                    fn some_function() -> Result<(), SomeErrStruct>;
+                }
+            }
+        }
+    }
+
+    // In Rust 1.79.0 dead_code warnings are issued for wrapped data in enums in spite of the enum
+    // having `#[repr(C)]`. `#[allow(unused)]` can be removed following resolution and release of this
+    // issue: https://github.com/rust-lang/rust/issues/126706
+    fn expected_rust_tokens() -> ExpectedRustTokens {
+        ExpectedRustTokens::Contains(quote! {
+            #[repr(C)]
+            pub enum ResultVoidAndSomeErrStruct{
+                #[allow(unused)]
+                Ok,
+                #[allow(unused)]
+                Err(__swift_bridge__SomeErrStruct),
+            }
+
+            #[export_name = "__swift_bridge__$some_function"]
+            pub extern "C" fn __swift_bridge__some_function() -> ResultVoidAndSomeErrStruct{
+                match super::some_function() {
+                    Ok(ok) => ResultVoidAndSomeErrStruct::Ok,
+                    Err(err) => ResultVoidAndSomeErrStruct::Err(err.into_ffi_repr()),
+                }
+            }
+        })
+    }
+
+    fn expected_swift_code() -> ExpectedSwiftCode {
+        ExpectedSwiftCode::ContainsAfterTrim(
+            r#"
+public func some_function() throws -> () {
+    try { let val = __swift_bridge__$some_function(); switch val.tag { case __swift_bridge__$ResultVoidAndSomeErrStruct$ResultOk: return case __swift_bridge__$ResultVoidAndSomeErrStruct$ResultErr: throw val.payload.err.intoSwiftRepr() default: fatalError() } }()
+}
+"#,
+        )
+    }
+
+    fn expected_c_header() -> ExpectedCHeader {
+        ExpectedCHeader::ContainsManyAfterTrim(vec![
+            r#"
+typedef enum __swift_bridge__$ResultVoidAndSomeErrStruct$Tag {__swift_bridge__$ResultVoidAndSomeErrStruct$ResultOk, __swift_bridge__$ResultVoidAndSomeErrStruct$ResultErr} __swift_bridge__$ResultVoidAndSomeErrStruct$Tag;
+union __swift_bridge__$ResultVoidAndSomeErrStruct$Fields {struct __swift_bridge__$SomeErrStruct err;};
+typedef struct __swift_bridge__$ResultVoidAndSomeErrStruct{__swift_bridge__$ResultVoidAndSomeErrStruct$Tag tag; union __swift_bridge__$ResultVoidAndSomeErrStruct$Fields payload;} __swift_bridge__$ResultVoidAndSomeErrStruct;
+"#,
+            r#"struct __swift_bridge__$ResultVoidAndSomeErrStruct __swift_bridge__$some_function(void)"#,
+        ])
+    }
+
+    #[test]
+    fn extern_rust_result_transparent_struct_type_and_opaque_rust_type() {
+        CodegenTest {
+            bridge_module: bridge_module_tokens().into(),
+            expected_rust_tokens: expected_rust_tokens(),
+            expected_swift_code: expected_swift_code(),
+            expected_c_header: expected_c_header(),
+        }
+        .test();
+    }
+}

--- a/crates/swift-bridge-ir/src/codegen/codegen_tests/result_codegen_tests.rs
+++ b/crates/swift-bridge-ir/src/codegen/codegen_tests/result_codegen_tests.rs
@@ -407,11 +407,15 @@ mod extern_rust_fn_return_result_opaque_rust_type_and_transparent_enum_type {
         }
     }
 
+    // In Rust 1.79.0 dead_code warnings are issued for wrapped data in enums in spite of the enum
+    // having `#[repr(C)]`.
     fn expected_rust_tokens() -> ExpectedRustTokens {
         ExpectedRustTokens::Contains(quote! {
             #[repr(C)]
             pub enum ResultSomeOkTypeAndSomeErrEnum{
+                #[allow(unused)]
                 Ok(*mut super::SomeOkType),
+                #[allow(unused)]
                 Err(__swift_bridge__SomeErrEnum),
             }
 
@@ -484,11 +488,14 @@ mod extern_rust_fn_return_result_transparent_enum_type_and_opaque_rust_type {
         }
     }
 
+    // Allows unused to avoid dead_code warnings in Rust 1.79.0 or later.
     fn expected_rust_tokens() -> ExpectedRustTokens {
         ExpectedRustTokens::Contains(quote! {
             #[repr(C)]
             pub enum ResultSomeOkEnumAndSomeErrType{
+                #[allow(unused)]
                 Ok(__swift_bridge__SomeOkEnum),
+                #[allow(unused)]
                 Err(*mut super::SomeErrType),
             }
 
@@ -558,11 +565,14 @@ mod extern_rust_fn_return_result_unit_type_and_transparent_enum_type {
         }
     }
 
+    // Allows unused to avoid dead_code warnings in Rust 1.79.0 or later.
     fn expected_rust_tokens() -> ExpectedRustTokens {
         ExpectedRustTokens::Contains(quote! {
             #[repr(C)]
             pub enum ResultVoidAndSomeErrEnum{
+                #[allow(unused)]
                 Ok,
+                #[allow(unused)]
                 Err(__swift_bridge__SomeErrEnum),
             }
 
@@ -628,12 +638,15 @@ mod extern_rust_fn_return_result_tuple_type_and_transparent_enum_type {
         }
     }
 
+    // Allows unused to avoid dead_code warnings in Rust 1.79.0 or later.
     fn expected_rust_tokens() -> ExpectedRustTokens {
         ExpectedRustTokens::ContainsMany(vec![
             quote! {
                 #[repr(C)]
                 pub enum ResultTupleI32U32AndSomeErrEnum{
+                    #[allow(unused)]
                     Ok(__swift_bridge__tuple_I32U32),
+                    #[allow(unused)]
                     Err(__swift_bridge__SomeErrEnum),
                 }
             },

--- a/crates/swift-integration-tests/src/result.rs
+++ b/crates/swift-integration-tests/src/result.rs
@@ -38,12 +38,14 @@ mod ffi {
     }
 
     #[swift_bridge(swift_repr = "struct")]
-    struct ResultTestTransparentStruct(pub String);
+    struct ResultTransparentStruct {
+        pub inner: String,
+    }
 
     extern "Rust" {
-        fn rust_func_returns_result_null_transparent_struct(
+        fn rust_func_return_result_null_transparent_struct(
             succeed: bool,
-        ) -> Result<(), ResultTestTransparentStruct>;
+        ) -> Result<(), ResultTransparentStruct>;
     }
 
     enum ResultTransparentEnum {
@@ -146,27 +148,29 @@ fn rust_func_return_result_unit_struct_opaque_rust(
     }
 }
 
-fn rust_func_returns_result_null_transparent_struct(
+fn rust_func_return_result_null_transparent_struct(
     succeed: bool,
-) -> Result<(), ffi::ResultTestTransparentStruct> {
+) -> Result<(), ffi::ResultTransparentStruct> {
     if succeed {
         Ok(())
     } else {
-        Err(ffi::ResultTestTransparentStruct("failed".to_string()))
+        Err(ffi::ResultTransparentStruct {
+            inner: "failed".to_string(),
+        })
     }
 }
 
-impl std::error::Error for ffi::ResultTestTransparentStruct {}
+impl std::error::Error for ffi::ResultTransparentStruct {}
 
-impl std::fmt::Debug for ffi::ResultTestTransparentStruct {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", self)
+impl std::fmt::Debug for ffi::ResultTransparentStruct {
+    fn fmt(&self, _f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        unreachable!("Debug impl was added to pass `Error: Debug + Display` type checking")
     }
 }
 
-impl std::fmt::Display for ffi::ResultTestTransparentStruct {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", self.0)
+impl std::fmt::Display for ffi::ResultTransparentStruct {
+    fn fmt(&self, _f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        unreachable!("Display impl was added to pass `Error: Debug + Display` type checking")
     }
 }
 

--- a/crates/swift-integration-tests/src/result.rs
+++ b/crates/swift-integration-tests/src/result.rs
@@ -1,8 +1,4 @@
 //! See also: crates/swift-bridge-ir/src/codegen/codegen_tests/result_codegen_tests.rs
-// This is a temporary workaround until https://github.com/chinedufn/swift-bridge/issues/270
-// is closed. When tests are compiled they have `-D warnings` (deny warnings) enabled, so
-// tests won't even compile unless this warning is ignored.
-#![allow(dead_code)]
 
 #[swift_bridge::bridge]
 mod ffi {
@@ -39,6 +35,15 @@ mod ffi {
         type ResultTestOpaqueSwiftType;
 
         fn val(&self) -> u32;
+    }
+
+    #[swift_bridge(swift_repr = "struct")]
+    struct ResultTestTransparentStruct(pub String);
+
+    extern "Rust" {
+        fn rust_func_returns_result_null_transparent_struct(
+            succeed: bool,
+        ) -> Result<(), ResultTestTransparentStruct>;
     }
 
     enum ResultTransparentEnum {
@@ -138,6 +143,30 @@ fn rust_func_return_result_unit_struct_opaque_rust(
         Ok(ffi::UnitStruct)
     } else {
         Err(ResultTestOpaqueRustType { val: 222 })
+    }
+}
+
+fn rust_func_returns_result_null_transparent_struct(
+    succeed: bool,
+) -> Result<(), ffi::ResultTestTransparentStruct> {
+    if succeed {
+        Ok(())
+    } else {
+        Err(ffi::ResultTestTransparentStruct("failed".to_string()))
+    }
+}
+
+impl std::error::Error for ffi::ResultTestTransparentStruct {}
+
+impl std::fmt::Debug for ffi::ResultTestTransparentStruct {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self)
+    }
+}
+
+impl std::fmt::Display for ffi::ResultTestTransparentStruct {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
     }
 }
 

